src/app/core/permissions/ability/ability.service.spec.ts

```diff
@@ -159,22 +198,29 @@ describe("AbilityService", () => {
       entity: new Config(Config.PERMISSION_KEY, rules),
       type: "update",
     });
+
+    //TODO: is the current implementation as a replaySubject rather than an explicit triggering after updating the ability really correct?
```

Comment:
Do you know the background of this functionality (and test), @TheSlimvReal ?

Reply:
Yes, the common use cases were: notify once it is there and then notify me whenever it gets updated. This is where the replay subject comes in handy. We do not need to check whether it already has been loaded. We just subscribe and in case it is already loaded we get notified instantly. Is there any problem with it?

Reply:
I think the update might trigger before the code in the service, actually updating the EntityAbility runs.
(And also, with the rework here this test somehow fails :wink: )

How about "manually" triggering updatedAbility after we have applied ability.update() ?

Reply:
Yes, it looks like the `async` call breaks this. As the ability updated is barely used I guess its fine to change that. I also just found `this.ability.on('update', () => {})` which allows to listen for updates that are published by casl itself. So maybe thats even easier.